gui/src/hooks/discord-presence.ts

```diff
@@ -0,0 +1,71 @@
+import { useCallback, useEffect } from 'react';
+import { useConfig } from './config';
+import { useInterval } from './timeout';
+import { useTrackers } from './tracker';
+import { invoke } from '@tauri-apps/api/core';
+import { warn } from '@/utils/logging';
+import { useLocalization } from '@fluent/react';
+
+export function useDiscordPresence() {
+  const { config } = useConfig();
+  const { useConnectedIMUTrackers } = useTrackers();
+  const { l10n } = useLocalization();
+  const imuTrackers = useConnectedIMUTrackers();
+
+  const updatePresence = useCallback(() => {
+    (async () => {
+      try {
+        if (await checkDiscordClient()) {
+          // If discord client exists, try updating presence
+          await updateDiscordPresence({
+            details: l10n.getString(
+              'settings-general-interface-discord_presence-message',
+              { amount: imuTrackers.length }
+            ),
+          });
+        } else {
+          // else, try creating a discord client
+          await createDiscordClient();
+        }
+      } catch (e) {
+        warn(`failed to update presence, error: ${e}`);
+      }
+    })();
+  }, [imuTrackers.length, l10n]);
+
+  // Update presence every 7 seconds
+  useInterval(updatePresence, config?.discordPresence ? 7000 : null);
```

Comment:
```suggestion
  // Update presence every 6.9 seconds
  useInterval(updatePresence, config?.discordPresence ? 6900 : null);
```
this would be way funnier